platform/mbed_error.h

```diff
@@ -60,7 +67,17 @@
 extern "C" {
 #endif
 
-void error(const char* format, ...);
+
+static inline void error(const char *format, ...) {
```

Comment:
Here's the footprint (additional 1.5k flash) with debug correctly optimized out and @pan-'s workaround for the fflushs in exit. There's no stdio being dragged in, but the format strings/arguments can't be garbage collected by the compiler:
```

Note: The savings will grow with the more calls to error
+--------------------------+-------+-------+------+
| Module                   | .text | .data | .bss |
+--------------------------+-------+-------+------+
| Misc                     |   348 |     0 | 1024 |
| features/storage         |    30 |     0 |  184 |
| hal                      |   232 |     8 |    0 |
| platform                 |   188 |    16 |    0 |
| rtos                     |    36 |     8 |    0 |
| rtos/rtx                 |  5766 |   100 | 6764 |
| targets/TARGET_Freescale |  4712 |    92 |  112 |
| Subtotals                | 11312 |   224 | 8084 |
+--------------------------+-------+-------+------+
```